### DISCUSSION TITLE
How to disconnect then reconnect or connect to another channel

### DISCUSSION THREAD

**Discussion by XcomReborn** [Bukk94/KickLib#6] [Q&A] [answered]
The following sample code, connects, then disconnects using the DisconnectAsync() method.

```
using KickLib.Interfaces;
using KickLib;
using KickLib.Client.Interfaces;
using KickLib.Client.Models.Args;
using KickLib.Client;
using KickLib.Client.Models.Events.Chatroom;
using KickLib.Models;
using static System.Net.Mime.MediaTypeNames;
using PusherClient;
using KickLib.Api;

Console.WriteLine("Starting!");

bool running = true;

var userName = "Xcom-Reborn";

IKickApi kickApi = new KickApi();

var channelInfo = await kickApi.Channels.GetChannelInfoAsync(userName);

IKickClient client = new KickClient();

client.OnMessage += delegate (object sender, ChatMessageEventArgs e)
{

    Console.WriteLine(e.Data.Content);

    if (e.Data.Content == "quit")
    {
        running = false;
    }

};


await client.ListenToChatRoomAsync(channelInfo.Chatroom.Id);

await client.ConnectAsync();

Console.WriteLine(client.IsConnected.ToString());

await client.DisconnectAsync();

Console.WriteLine(client.IsConnected.ToString());

channelInfo = await kickApi.Channels.GetChannelInfoAsync(userName);

await client.ListenToChatRoomAsync(channelInfo.Chatroom.Id);

await client.ConnectAsync();

Console.WriteLine(client.IsConnected.ToString());


while (running)
{
    await Task.Delay(500);
}

Console.WriteLine("Exiting...");

```

This throws a timeout exception the second time await kickApi.Channels.GetChannelInfoAsync(userName); is called. I've tried doing it with different channel names but same issue.

Any ideas ?

**Comment by Bukk94** [comment, marked as answer]
Unfortunately, if this is happening it means that Kick's protection system got much more aggressive. Based on my testing, first few requests will go through but then you are blocked by CAPTCHA.
To fetch a data, it uses C# Puppeteer with stealth and Anonymize UA plugins. The only option is to customize the plugins more to avoid detection or find another way of accessing Kick's Cloudflare-protected API.

**Reply by XcomReborn** [reply]
Ok no problem, I understand the frustration with dealing with kick and their non public, non documented API. I'm grateful for you library because I was waiting for over a year and a half for them to releases an actual public API so I could implement my text to speech program and others. My current project now works (Its not pretty but it works) thanks to your efforts.

Many thanks.
